wsproxy/websocket_proxy.go

```diff
@@ -6,15 +6,75 @@ import (
 	"net/http"
 	"strings"
 
-	log "github.com/Sirupsen/logrus"
+	"github.com/Sirupsen/logrus"
 	"github.com/gorilla/websocket"
 	"golang.org/x/net/context"
 )
 
-var (
-	MethodOverrideParam = "method"
-	TokenCookieName     = "token"
-)
+// MethodOverrideParam defines the special URL parameter that is translated into the subsequent proxied streaming http request's method.
+var MethodOverrideParam = "method"
+
+// TokenCookieName defines the cookie name that is translated to an 'Authorization: Bearer' header in the streaming http request's headers.
+//
+// Deprecated: it is preferable to use the Options parameters to WebSocketProxy to supply parameters.
+var TokenCookieName = "token"
+
+// RequestMutatorFunc can modify a outgoing proxied request before it is made.
+type RequestMutatorFunc func(incoming *http.Request, outgoing *http.Request)
```

Comment:
What use cases do we imagine for this function? The signature feels very C like, any chance we could have the mutated request returned from the function instead?